client/homebrew/editor/snippetbar/snippetbar.jsx

```diff
@@ -151,12 +164,18 @@ const Snippetbar = createClass({
 		return this.props.updateBrew(item);
 	},
 
+	toggleHistoryMenu : function(){
+		this.setState({
+			showHistory : !this.state.showHistory
+		});
+	},
+
 	renderHistoryItems : function() {
-		const historyItems = getHistoryItems(this.props.brew);
+		if(this.state.historyItems.length == 0) return;
```

Comment:
is `historyItems` redundant with `historyExists`? HistoryExists also requires at least one entry is in `historyItems`. Seems `historyExists` could be removed altogether and just use `historyItems` like you do here?

Reply:
Correct, I started out just having `historyExists` is a boolean function, but I had been considering using a `historyItems.length` declaration instead. However, as `loadHistory` should return a non-zero length Array every time, it should be slightly more complicated than just a `length` check.

Reply:
If `loadHistory` returns a non-zero length Array everytime, is this heck for `historyItems.length == 0` ever going to be true?

Reply:
Only if it hasn't populated it yet... in which case the renderHistoryItems function should not be accessible, as it is locked behind the historyExists function.
This is implicit in the way the render function is called, but I think I'll add it as an explicit guard clause inside the renderHistoryItems function.